src/api.c

```diff
@@ -6118,3 +6119,49 @@ const struct cgroup_library_version *cgroup_version(void)
 {
 	return &library_version;
 }
+
+/**
+ * Finds the current cgroup setup mode (legacy/unified/hybrid).
+ * Returns unknown of failure and setup mode on success.
+ */
+enum cg_setup_mode_t cgroup_setup_mode(void)
+{
+#define CGROUP2_SUPER_MAGIC	0x63677270
+#define CGROUP_SUPER_MAGIC	0x27E0EB
+
+	unsigned int cg_setup_mode_bitmask = 0U;
+	enum cg_setup_mode_t setup_mode;
+	struct statfs cgrp_buf;
+	int i, ret = 0;
+
+	setup_mode = CGROUP_UNKNOWN;
+
+	pthread_rwlock_wrlock(&cg_mount_table_lock);
+	for (i = 0; cg_mount_table[i].name[0] != '\0'; i++) {
+		ret = statfs(cg_mount_table[i].mount.path, &cgrp_buf);
+		if (ret) {
+			setup_mode = CGROUP_UNKNOWN;
+			cgroup_dbg("Failed to get stats of '%s'\n", cg_mount_table[i].mount.path);
```

Comment:
nitpick - should we promote this to `cgroup_err()` since this is a failure case?

Reply:
Sure, lets error out in this unlikely event().